pkg/minikube/cluster/cluster.go

```diff
@@ -122,12 +132,27 @@ func DeleteHost(api libmachine.API) error {
 	if err != nil {
 		return err
 	}
+
+	if err = unregister(host); err != nil {
+		return err
+	}
+
 	m := util.MultiError{}
 	m.Collect(host.Driver.Remove())
 	m.Collect(api.Remove(constants.MachineName))
 	return m.ToError()
 }
 
+// Un-register host VM
+func unregister(host *host.Host) error {
+	if drivers.MachineInState(host.Driver, state.Running)() {
```

Comment:
> Agree,

Cool :-) Shall we create a new issue for this or do you want to continue on this pull request?

> should we use config file which we create at the beginning for this purpose or create a another one

I always thought about re-suing the existing one. Maybe we can start with that and see how it goes. If it creates problems we can always create a separate one. 

Basically I thought we read the config file as in _root.go_ (we already do as part of the Viper config) and keep the landed config in memory behind some Config interface. When we modify the config we flush. For this particulate case we would write a _isRegisteredVM_ flag. Maybe we put this into a nested _currentVM_ context. We will also need the current _oc_ to be used persistent in there. This is required for the _oc env_ command as well as for the registry exposure (issue #254).